### DISCUSSION TITLE
Google Cloud Managed Service for Prometheus support

### DISCUSSION THREAD

**Discussion by danielkucera** [grafana/grafana#46102] [Data Sources]
Last year, Google has introduced Managed Service for Prometheus which allows to query metrics using Prometheus API.
One can use the standard Prometheus API but Authentication token needs to be used, see:
https://cloud.google.com/stackdriver/docs/managed-prometheus/query#api-prometheus

It should be fairly easy to extend current Prometheus datasource plugin to get the token from:
http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token

and add it to request header

**Comment by clarkmcc** [comment]
Any chance this will be supported?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #83001. Feel free to continue the discussion around this there. Thank you!
